src/Logic.hs

```diff
@@ -92,4 +97,56 @@ transformGame (EventKey (SpecialKey KeyUp) Up _ _) game = performMove TopMov gam
 transformGame (EventKey (SpecialKey KeyDown) Up _ _) game = performMove DownMov game
 transformGame (EventKey (SpecialKey KeyLeft) Up _ _) game = performMove LeftMov game
 transformGame (EventKey (SpecialKey KeyRight) Up _ _) game = performMove RightMov game
-transformGame _ game = game
\ No newline at end of file
+transformGame (EventKey (Char 'r') Up _ _) game = reinitializeGame game
+
+transformGame _ game = game
+
+mergeBoard :: [[Cell]] -> ([[Cell]], Int)
+mergeBoard [] = ([], 0)
+mergeBoard (x : xs) = ((merged ++ padding) : mergedTail, score + scoreTail)
+  where
+    (merged, score) = mergeCells x
+    padding = replicate (length x - length merged) Empty
+    (mergedTail, scoreTail) = mergeBoard xs
+
+merge :: [[Cell]] -> Direction -> ([[Cell]], Int)
+merge [] _ = ([], 0)
+merge cell d = (move, score)
+  where
+    (newCells, score) = mergeBoard cell
+    move = case d of
+      TopMov -> transpose newCells
+      DownMov -> transpose $ map reverse newCells
+      LeftMov -> newCells
+      RightMov -> map reverse newCells
+
+-- This function restarts the game board, score and random generator
+reinitializeGame :: Game -> Game
+reinitializeGame game = game { gameBoard = newBoard, gameScore =0, gameStdGen = newStdGen }
+  where
+    (newBoard, newStdGen) = generateTwoRandomCells (gameStdGen game) (createEmptyBoard n)
```

Comment:
Change  the name of the variable "newStdGen"

This binding for ‘newStdGen’ shadows the existing binding
  imported from ‘System.Random’ at /home/fundacion/Clases/programacion/Proyects/HaskellWork/game2048/src/Logic.hs:8:1-20typecheck(-Wname-shadowing)
newStdGen :: StdGen